backend/controller/evaljob_controller.go

```diff
@@ -0,0 +1,568 @@
+		controllerutil.AddFinalizer(job, lmevalservicev1beta1.FinalizerName)
+		if err := r.Update(ctx, job); err != nil {
+			log.Error(err, "unable to update finalizer")
+			return ctrl.Result{}, err
+		}
+		r.Recorder.Event(job, "Normal", "AttachFinalizer",
+			fmt.Sprintf("added the finalizer to the EvalJob %s in namespace %s",
+				job.Name,
+				job.Namespace))
+		// Since finalizers were updated. Need to fetch the new EvalJob
+		// End the current reconsile and get revisioned job in next reconsile
+		return ctrl.Result{}, nil
+	}
+
+	// construct a new pod and create a pod for the job
+	currentTime := v1.Now()
+	pod := r.createPod(job)
+	if err := r.Create(ctx, pod, &client.CreateOptions{}); err != nil {
+		// Failed to create the pod. Mark the status as complete with failed
+		job.Status.State = lmevalservicev1beta1.CompleteJobState
+		job.Status.Reason = lmevalservicev1beta1.FailedReason
+		job.Status.Message = err.Error()
+		if err := r.Status().Update(ctx, job); err != nil {
+			log.Error(err, "unable to update EvalJob status for pod creation failure")
+		}
+		log.Error(err, "Failed to create pod for the EvalJob", "name", job.Name)
+		return ctrl.Result{}, err
+	}
+
+	// Create the pod successfully. Wait for the driver to update the status
+	job.Status.State = lmevalservicev1beta1.ScheduledJobState
+	job.Status.PodName = pod.Name
+	job.Status.LastScheduleTime = &currentTime
+	if err := r.Status().Update(ctx, job); err != nil {
+		log.Error(err, "unable to update EvalJob status (pod creation done)")
+		return ctrl.Result{}, err
+	}
+	r.Recorder.Event(job, "Normal", "PodCreation",
+		fmt.Sprintf("the EvalJob %s in namespace %s created a pod",
+			job.Name,
+			job.Namespace))
+	log.Info("Successfully create a Pod for the Job")
+	// Check the pod after 10 seconds
+	return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+}
+
+func (r *EvalJobReconciler) checkScheduledPod(ctx context.Context, log logr.Logger, job *lmevalservicev1beta1.EvalJob) (ctrl.Result, error) {
+	pod, err := r.getPod(ctx, job)
+	if err != nil {
+		// a weird state, someone delete the corresponding pod? mark this as CompleteJobState
+		// with error message
+		job.Status.State = lmevalservicev1beta1.CompleteJobState
+		job.Status.Reason = lmevalservicev1beta1.FailedReason
+		job.Status.Message = err.Error()
+		if err := r.Status().Update(ctx, job); err != nil {
+			log.Error(err, "unable to update EvalJob status", "state", job.Status.State)
+			return ctrl.Result{}, err
+		}
+		r.Recorder.Event(job, "Warning", "PodMising",
+			fmt.Sprintf("the pod for the EvalJob %s in namespace %s is gone",
+				job.Name,
+				job.Namespace))
+		log.Error(err, "since the job's pod is gone, mark the job as complete with error result.")
+		return ctrl.Result{}, err
+	}
+
+	if pod.Status.ContainerStatuses == nil {
+		// wait for the pod to initialize and run the containers
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
```

Comment:
Worth making this requeue timing configurable?

Reply:
yes, I will put it into the `ConfigMap`